src/http/web_socket.cr

```diff
@@ -110,38 +110,38 @@ class HTTP::WebSocket
       end
 
       case info.opcode
-      when Protocol::Opcode::PING
+      when .ping?
         @current_message.write @buffer[0, info.size]
         if info.final
           message = @current_message.to_s
           @on_ping.try &.call(message)
           pong(message) unless closed?
           @current_message.clear
         end
-      when Protocol::Opcode::PONG
+      when .pong?
         @current_message.write @buffer[0, info.size]
         if info.final
           @on_pong.try &.call(@current_message.to_s)
           @current_message.clear
         end
-      when Protocol::Opcode::TEXT
+      when .text?
         @current_message.write @buffer[0, info.size]
         if info.final
           @on_message.try &.call(@current_message.to_s)
           @current_message.clear
         end
-      when Protocol::Opcode::BINARY
+      when .binary?
         @current_message.write @buffer[0, info.size]
         if info.final
           @on_binary.try &.call(@current_message.to_slice)
           @current_message.clear
         end
-      when Protocol::Opcode::CLOSE
+      when .close?
         @current_message.write @buffer[0, info.size]
         if info.final
           message = @current_message.to_s
           @on_close.try &.call(message)
-          close(message) unless closed?
+          close(message.presence)
```

Comment:
Why change behaviour here?

Reply:
Why would we want to pass an empty string here?

Reply:
Why would you convert empty strings into `nil`?